pkg/controllers/resources/pods/translate/translator.go

```diff
@@ -172,15 +174,26 @@ func (t *translator) Translate(ctx *synccontext.SyncContext, vPod *corev1.Pod, s
 	pPod.Spec.AutomountServiceAccountToken = &False
 	pPod.Spec.EnableServiceLinks = &False
 
-	// check if priority classes are enabled
-	if !t.priorityClassesEnabled {
-		pPod.Spec.PriorityClassName = ""
-		pPod.Spec.Priority = nil
-	} else if pPod.Spec.PriorityClassName != "" {
+	// by default the priority class would be synced as-is to the host
+	// this will readily work when config.Sync.FromHost.PriorityClasses.Enabled is true
+	// however, in case when config.Sync.ToHost.PriorityClasses.Enabled is true, we
+	// need to translate the priorityClassName before syncing to host
+	if t.priorityClassesSyncEnabled && pPod.Spec.PriorityClassName != "" {
```

Comment:
I was trying to think of a little refactor so that we're not checking `t.priorityClassesSyncEnabled` in both the if and else if, but it does read better to check that both are disabled.  Very nit suggestion, but maybe naming the conditionals in variables, like `allSyncingDisabled := !t.priorityClassesSyncEnabled && !t.hostPriorityClassesSyncEnabled`.

Reply:
@zerbitx I have created a separate PR for this [here](https://github.com/loft-sh/vcluster/pull/2498)